src/ai2_internal/bibentry_detection_predictor/interface.py

```diff
@@ -98,13 +98,18 @@ def predict_one(self, inst: Instance) -> Prediction:
         doc.annotate_images(images)
         doc.annotate(vila_span_groups=[sg.to_mmda() for sg in inst.vila_span_groups])
 
-        processed_bib_entry_box_groups, original_box_groups = self._predictor.predict(doc, self._config.BIB_ENTRY_DETECTION_MIN_VILA_BIB_ROWS)
+        processed_bib_entry_box_groups, original_box_groups = self._predictor.predict(
+            doc,
+            self._config.BIB_ENTRY_DETECTION_MIN_VILA_BIB_ROWS
+        )
 
         # generate SpanGroups
         if len(processed_bib_entry_box_groups) > 0:
             doc.annotate(bib_entries=processed_bib_entry_box_groups)
             prediction = Prediction(
-                bib_entries=[api.SpanGroup.from_mmda(sg) for sg in doc.bib_entries],
+                # filter out span-less SpanGroups
+                bib_entries=[api.SpanGroup.from_mmda(sg) for sg in doc.bib_entries if len(sg.spans) != 0],
```

Comment:
can you remind me why we end up producing spanless SpanGroups that we then have to filter out?

ok, from your comments above, I remember that we end up with overlapping bib boxes, and then the spans can only be assigned to one of them. But do you have a sense of what causes us to draw the bib boxes badly?

Reply:
My sense of what causes us to draw the bib boxes badly is not a good understanding but - I know it has to do with the original model output boxes simply being "bad" - it's possible that increasing the threshold of model score for boxes returned by the model might reduce the amount that this happens but I haven't looked closely at enough examples to know for sure if that would help or make things worse by giving us fewer boxes overall - I'm sure there's a sweet spot but I'm not sure how to find it! Simply filtering these ones out seems like the best thing to do for now.

Reply:
👍